### ISSUE TITLE
IOS XE show clns (isis.py) missing key adjacency, and mismatched adjacent routers

### ISSUE THREAD

**Issue by filippohronsky** [CiscoTestAutomation/genieparser#170]
Hello, there is an issue with genie parser for isis.
output file for ios xe device is damaged in a way that values are mixed.
![Snímka obrazovky 2020-06-17 o 23 18 13](https://user-images.githubusercontent.com/24731316/84952240-0030e500-b0f2-11ea-9727-3b79eefefaf5.png)

sh isis neighbors 

```
Tag core:
System Id       Type Interface     IP Address      State Holdtime Circuit Id
RTR3            L1   Te0/2/0       19.1.1.0      UP    27       00
RT2             L1   Te0/2/1       19.8.1.2      UP    25       0C
RTR4            L1   Te0/3/0       98.1.11.9      UP    26       01
RT1             L1   Gi0/0/7       8.18.10.130     UP    24       02
```

How can I find the parser error, and what should damage parser input to show this error?
Thank you.
Filip

**Comment by lubnarasheed** [comment]
Hi Filip,

Thanks for reporting this issue. @thaixuann will investigate and update this thread.

Regards,
Lubna Rasheed

**Comment by thaixuann** [comment]
Hello @filippohronsky ,
Per the output that you provided, the parser parsed correctly what given in the output. If you ran this parser and it error out, then we will fix it.
You may need to use different commands to parse the adjacency that you said missing?
This is the list of all parsers currently available to use: https://pubhub.devnetcloud.com/media/genie-feature-browser/docs/#/parsers

Please let me know if you need anything else.

Thanks,
--Thai.

**Comment by filippohronsky** [comment]
Hello Thai,
The output from CLI console shows four adjacent neighbors on four interfaces.
The parser output in JSON shows three adjacent neighbors on four interfaces.

Somehow there is incorrect JSON output from accurate CLI data.

CLI Ouput.
+++ DS_A903: executing command 'show clns neighbors detail' +++
show clns neighbors detail
Tag core:
System Id       Interface     SNPA                State  Holdtime  Type Protocol
RT3         Te0/2/0       6c71.0d38.0979      Up     28        L1   IS-IS
  Area Address(es): 01
  IP Address(es):  98.1.1.0*
  Uptime: 1w6d
  NSF capable
  BFD enabled: (MTID:0, ipv4)
  Interface name: TenGigabitEthernet0/2/0
RT2         Te0/2/1       7c21.0d05.701b      Up     26        L1   IS-IS
  Area Address(es): 01
  IP Address(es):  18.18.1.12*
  Uptime: 2w0d
  NSF capable
  BFD enabled: (MTID:0, ipv4)
  Interface name: TenGigabitEthernet0/2/1
1K1A     Te0/3/0       3c51.0ebc.4980      Up     25        L1   IS-IS //missing neighbor, instead RT2 shows in JSON.
  Area Address(es): 01
  IP Address(es):  19.18.51.9*
  Uptime: 2w0d
  NSF capable
  BFD enabled: (MTID:0, ipv4)
  Interface name: TenGigabitEthernet0/3/0
RT1          Gi0/0/7       000c.29a0.59ac      Up     28        L1   IS-IS
  Area Address(es): 01
  IP Address(es):  8.18.1.113*
  Uptime: 2w0d
  NSF capable
  BFD enabled: (MTID:0, ipv4)
  Interface name: GigabitEthernet0/0/7
DS_A903#

[cid:image001.png@01D6458A.671178D0]


<filipp>


Od: THAI NGUYEN <notifications@github.com>
Príjemca odpovede: CiscoTestAutomation/genieparser <reply@reply.github.com>
Dátum: štvrtok, 18. júna 2020, 3:53
Komu: CiscoTestAutomation/genieparser <genieparser@noreply.github.com>
Kópia: Pohronsky Filip <Pohronsky@ditec.sk>, Mention <mention@noreply.github.com>
Predmet: Re: [CiscoTestAutomation/genieparser] IOS XE show clns (isis.py) missing key adjacency, and mismatched adjacent routers (#170)


Hello @filippohronsky<https://github.com/filippohronsky> ,
Per the output that you provided, the parser parsed correctly what given in the output. If you ran this parser and it error out, then we will fix it.
You may need to use different commands to parse the adjacency that you said missing?
This is the list of all parsers currently available to use: https://pubhub.devnetcloud.com/media/genie-feature-browser/docs/#/parsers

Please let me know if you need anything else.

Thanks,
--Thai.

**Comment by thaixuann** [comment]
Hi @filippohronsky ,
Thanks for the output. I am looking into it and will get back to you.

Thanks,
--Thai.

**Comment by thaixuann** [comment]
Hello @filippohronsky ,
We released v20.6 yesterday, which included your fix.

`pip install --upgrade ats[full]`

Please try and let's know.

Thanks,
--Thai.

**Comment by thaixuann** [comment]
Hello @filippohronsky ,
Please let us know if there is anything else by open a new case. I will close this for now.

Thanks,
--Thai.
